aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/service/AWSS3StorageService.kt

```diff
@@ -85,6 +88,30 @@ internal class AWSS3StorageService(
         return URL(presignUrlRequest.url.toString())
     }
 
+    /**
+     * Validate if S3 object exists for the given key.
+     * Throws StorageException if NoSuchKey S3 client exception is caught.
+     * @param serviceKey S3 service key
+     */
+    override fun validateObjectExists(serviceKey: String) {
```

Comment:
lets not try catch here. Instead, lets catch when making the service call. This pattern is used throughout the rest of Amplifiy.

Reply:
try/catch removed in favor of catching at the operation level

Reply:
The reason we're using try-catch here is because the implementation requires it: we're utilizing S3's `headObject` API and catching `NotFound`, which is a Kotlin SDK type. This leverages the SDK to validate if the object exists.

If we don't catch the exception, we'll expose the abstraction of **StorageService**,  (see the import of `aws.sdk.kotlin.services.s3.model.NotFound` in AWSS3StoragePathGetPresignedUrlOperation.kt)

One alternative I can think of is try-catching here and changing `validateObjectExists` to return a bool indicating whether the object exists or not for the caller to handle

Reply:
Not a bad idea, but we would still need try/catch for the other failure scenarios. I see what you are saying in that NotFound is really as success case of validating that the object didn't exist, and not a true failure.

Reply:
I've refactored the exception handling to isolate `aws.sdk.kotlin.services.s3.model.NotFound` to `StorageService`.

Reply:
Awesome, thanks for taking the time to address my consideration